### PR TITLE
SAK-43838: SAMIGO - Error when editing Matching questions

### PR DESCRIPTION
**Pull request by FerAlvarez95**

### CODE REVIEW

**Comment by jesusmmp** [comment]
LGTM!

**Comment by adrianfish** [comment]
You've used the wrong SAK number in the commit message. It should be SAK-43838.

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ItemAddListener.java [hunk 1]
```diff
@@ -1123,26 +1123,27 @@ else if (itemauthor.getTarget() != null && (itemauthor.getTarget().equals("samba
...
+	        }
+	
+	        // if assign to pool, add the item to the pool
+	        if (StringUtils.isNotEmpty(bean.getSelectedPool())
+	                && !qpdelegate.hasItem(item.getItemIdString(), Long.valueOf(bean.getSelectedPool()))) {
+	          // if the item is already in the pool then do not add.
+	          // This is a scenario where the item might already be in the pool:
+	          // create an item in an assessemnt and assign it to p1
+	          // copy item from p1 to p2. 
+	          // now the item is already in p2. and if you want to edit the original item in the assessment, and reassign it to p2, you will get a duplicate error.
```
Comment (adrianfish) [inline samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ItemAddListener.java:1143]: Do these comments make any sense, now. If not, just remove them

**Review by adrianfish — CHANGES_REQUESTED** [review]
You need to change the SAK number in the commit. If that means closing this PR and submitting it again, no problem. Also, those comments don't make sense before your change.

File: samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ItemAddListener.java (hunk 1 above)
Comment (FerAlvarez95) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ItemAddListener.java]: Comments removed. I did another commit with the right SAK, changed the title too. Thanks!

Comment (jesusmmp) [reply samigo/samigo-app/src/java/org/sakaiproject/tool/assessment/ui/listener/author/ItemAddListener.java]: @adrianfish, this is ready to merge. Thanks!

**Review by juanjmerono — APPROVED** [review]

**Pull request merged**